KomaMRIBase/src/datatypes/phantom/motion/SimpleMotion.jl

```diff
@@ -0,0 +1,70 @@
+# ------ SimpleMotionTypes
+abstract type SimpleMotionType{T<:Real} end
+
+"""
+Simple Motion
+
+x = x + ux
+y = y + uy
+z = z + uz
+
+"""
+# -------- SimpleMotion
+mutable struct SimpleMotion{T<:Real} <: MotionModel{T}
+    types::Vector{<:SimpleMotionType{T}}
+end
+
+Base.getindex(motion::SimpleMotion, p::Union{AbstractRange,AbstractVector,Colon}) = motion
+Base.getindex(motion::SimpleMotion, p::Union{AbstractRange,AbstractVector,Colon}, 
+                                    q::Union{AbstractRange,AbstractVector,Colon}) = motion
+
+function get_spin_coords(motion::SimpleMotion{T}, x::AbstractVector{T}, y::AbstractVector{T}, z::AbstractVector{T}, t::AbstractArray{T}) where {T<:Real}
+    xi, yi, zi = x, y, z
+    composable_motions = motion.types[is_composable.(motion.types)]
+    sort!(composable_motions, by= m->m.ti)
+    for motion in composable_motions
+        xi, yi, zi = position_x(motion, xi, yi, zi, t), position_y(motion, xi, yi, zi, t), position_z(motion, xi, yi, zi, t)
+    end
+    xt = xi .+ reduce(.+, map((type) -> !is_composable(type) ? displacement_x(type, x, y, z, t) : 0, motion.types))
+    yt = yi .+ reduce(.+, map((type) -> !is_composable(type) ? displacement_y(type, x, y, z, t) : 0, motion.types))
+    zt = zi .+ reduce(.+, map((type) -> !is_composable(type) ? displacement_z(type, x, y, z, t) : 0, motion.types))
+    return xt, yt, zt
+end
```

Comment:
```suggestion
function get_spin_coords(motion::SimpleMotion{T}, x::AbstractVector{T}, y::AbstractVector{T}, z::AbstractVector{T}, t::AbstractArray{T}) where {T<:Real}
    xi, yi, zi = x, y, z
    composable_motions = motion.types[is_composable.(motion.types)]
    sort!(composable_motions, by= m->m.ti)
    for motion in composable_motions
        xi, yi, zi = position_x(motion, xi, yi, zi, t), position_y(motion, xi, yi, zi, t), position_z(motion, xi, yi, zi, t)
    end
    additive_motions = motion.types[!is_composable.(motion.types)]
    xt = xi .+ reduce(.+, map(type -> displacement_x(type, x, y, z, t), additive_motions))
    yt = yi .+ reduce(.+, map(type -> displacement_y(type, x, y, z, t), additive_motions))
    zt = zi .+ reduce(.+, map(type -> displacement_z(type, x, y, z, t), additive_motions))
    return xt, yt, zt
end
```
I do not like that we need to define unnecessary variables in the structs, like `ti` for periodic motions. I suggest creating a function `start()` and `end()` that applies to all simple motions. Maybe `ti` and `tf` can be renamed `start` and `end`. You already do this with `get_time_nodes`, and have the start defined for all of them. So `get_time_nodes(motion_type::SimpleMotionType) = [start(motion); ...; end(motion)]`, and you define `start` and `end` instead.